src/Tasks/ManifestUtil/PathUtil.cs

```diff
@@ -162,7 +162,9 @@ public static bool IsPEFile(string path)
             byte[] buffer = new byte[2];
             using (Stream s = File.Open(path, FileMode.Open, FileAccess.Read, FileShare.Read))
             {
+#pragma warning disable CA2022 // Avoid inexact read with 'Stream.Read'
                 s.Read(buffer, 0, 2);
+#pragma warning restore CA2022 // Avoid inexact read with 'Stream.Read'
```

Comment:
Should we switch to `ReadExactly` and polyfill it on net472?

Reply:
For the scnarios where only 2 bytes are read, I decided to leave the s.Read and rely on the xml parser and proper exception handling. 
The plan is to use `ReadExactly` for cases when reading the files from 0 to filLength. and disable warning for net472.